diff --git a/src/datawizard/reduction.c b/src/datawizard/reduction.c
index ed44c6c65a..3164175265 100644
--- a/src/datawizard/reduction.c
+++ b/src/datawizard/reduction.c
@@ -66,8 +66,7 @@ void starpu_data_set_reduction_methods_with_args(starpu_data_handle_t handle, st
 	{
 		/* make sure that the flags are applied to the children as well */
 		starpu_data_handle_t child_handle = starpu_data_get_child(handle, child);
-		if (child_handle->nchildren > 0)
-			starpu_data_set_reduction_methods_with_args(child_handle, redux_cl, redux_cl_arg, init_cl, init_cl_arg);
+		starpu_data_set_reduction_methods_with_args(child_handle, redux_cl, redux_cl_arg, init_cl, init_cl_arg);
 	}
 
 	handle->redux_cl = redux_cl;
