diff --git a/include/starpu_data_interfaces.h b/include/starpu_data_interfaces.h
index 294b13d3b3..cf0705180a 100644
--- a/include/starpu_data_interfaces.h
+++ b/include/starpu_data_interfaces.h
@@ -1207,6 +1207,24 @@ size_t starpu_matrix_get_allocsize(starpu_data_handle_t handle);
 		STARPU_MATRIX_CHECK(interface);                               \
 		(((struct starpu_matrix_interface *)(interface))->allocsize); \
 	})
+#define STARPU_MATRIX_SET_NX(interface, newnx)                                   \
+	do {                                                                     \
+		STARPU_MATRIX_CHECK(interface);                                  \
+		(((struct starpu_matrix_interface *)(interface))->nx) = (newnx); \
+	}                                                                        \
+	while (0)
+#define STARPU_MATRIX_SET_NY(interface, newny)                                   \
+	do {                                                                     \
+		STARPU_MATRIX_CHECK(interface);                                  \
+		(((struct starpu_matrix_interface *)(interface))->ny) = (newny); \
+	}                                                                        \
+	while (0)
+#define STARPU_MATRIX_SET_LD(interface, newld)                                   \
+	do {                                                                     \
+		STARPU_MATRIX_CHECK(interface);                                  \
+		(((struct starpu_matrix_interface *)(interface))->ld) = (newld); \
+	}                                                                        \
+	while (0)
 #else
 /**
    Return a pointer to the matrix designated by \p interface, valid
@@ -1250,39 +1268,23 @@ size_t starpu_matrix_get_allocsize(starpu_data_handle_t handle);
    Return the allocated size of the matrix designated by \p interface.
 */
 #define STARPU_MATRIX_GET_ALLOCSIZE(interface)	(((struct starpu_matrix_interface *)(interface))->allocsize)
-#endif
-
 /**
    Set the number of elements on the x-axis of the matrix
    designated by \p interface.
 */
-#define STARPU_MATRIX_SET_NX(interface, newnx)                                   \
-	do {                                                                     \
-		STARPU_MATRIX_CHECK(interface);                                  \
-		(((struct starpu_matrix_interface *)(interface))->nx) = (newnx); \
-	}                                                                        \
-	while (0)
+#define STARPU_MATRIX_SET_NX(interface, newnx) (((struct starpu_matrix_interface *)(interface))->nx) = (newnx);
 /**
    Set the number of elements on the y-axis of the matrix
    designated by \p interface.
 */
-#define STARPU_MATRIX_SET_NY(interface, newny)                                   \
-	do {                                                                     \
-		STARPU_MATRIX_CHECK(interface);                                  \
-		(((struct starpu_matrix_interface *)(interface))->ny) = (newny); \
-	}                                                                        \
-	while (0)
+#define STARPU_MATRIX_SET_NY(interface, newny) (((struct starpu_matrix_interface *)(interface))->ny) = (newny);
 /**
    Set the number of elements between each row of the matrix
    designated by \p interface. May be set to the same value as nx when there is
    no padding.
 */
-#define STARPU_MATRIX_SET_LD(interface, newld)                                   \
-	do {                                                                     \
-		STARPU_MATRIX_CHECK(interface);                                  \
-		(((struct starpu_matrix_interface *)(interface))->ld) = (newld); \
-	}                                                                        \
-	while (0)
+#define STARPU_MATRIX_SET_LD(interface, newld) (((struct starpu_matrix_interface *)(interface))->ld) = (newld);
+#endif
 
 /** @} */
 
