diff --git a/src/datawizard/interfaces/matrix_filters.c b/src/datawizard/interfaces/matrix_filters.c
index 840932bb70..0fbf7c0762 100644
--- a/src/datawizard/interfaces/matrix_filters.c
+++ b/src/datawizard/interfaces/matrix_filters.c
@@ -149,7 +149,7 @@ void starpu_matrix_filter_pick_vector_y(void *father_interface, void *child_inte
 
 	uintptr_t chunk_pos = (uintptr_t)f->filter_arg_ptr;
 
-	STARPU_ASSERT_MSG(nchunks <= nx, "cannot get %u vectors", nchunks);
+	STARPU_ASSERT_MSG(nchunks <= ny, "cannot get %u vectors", nchunks);
 	STARPU_ASSERT_MSG((chunk_pos + id) < ny, "the chosen vector should be in the matrix");
 
 	size_t offset = (chunk_pos + id) * blocksize * elemsize;
